relay-server/src/actors/processor.rs

```diff
@@ -1105,10 +1117,8 @@ impl EnvelopeProcessorService {
 
         match result {
             Ok(response) => {
-                if let Some(managed_envelope) = response.envelope {
-                    self.inner.envelope_manager.send(SubmitEnvelope {
-                        envelope: managed_envelope,
-                    })
+                if let Some(envelope) = response.envelope {
+                    self.handle_submit_envelope(SubmitEnvelope { envelope });
```

Comment:
Should we instead send this message to ourselves? 

Metrics collected in the "event loop" for the `SubmitEnvelope` are incorrect now.
It's a little awkward since we don't currently have our own address.

Reply:
Services cannot send messages to themselves yet, we don't expose the address for that internally. I was thinking about this when writing it, but in a way envelope processing now simply also contains serializing the envelope. Otherwise, the actual submission still happens in another service. So, I think this is fine to leave as-is.